src/Compilers/CSharp/Portable/Symbols/AsyncKind.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal enum AsyncKind : byte
+    {
+        NotAsync = 0,
+        Invalid,
+        Void,
+        Task,
```

Comment:
I'd rename these... the hardest thing in this feature was recognizing whether we're talking about "Task" specifically or "non-generic Tasklike", and likewise about "Task<T>" specifically or "generic Tasklike".

In this case, the code uses AsyncKind.Task for all non-generic tasklikes, and uses ASyncKind.Task_T for all generic tasklikes. So I'd hope for a name that indicates this :)